iceweather/weather.py

```diff
@@ -62,21 +63,47 @@ def _distance(loc1: Tuple[float, float], loc2: Tuple[float, float]) -> float:
     return _EARTH_RADIUS * c
 
 
-def _api_call(url: str) -> Dict:
-    """ Use requests to call the apis.is weather API. """
+_ArgType = Union[int, str, Iterable[Union[int, str]]]
+
+
+def _arg_to_str_list(arg: _ArgType) -> List[str]:
+    """Helper function, converts argument to a list of strings."""
+    t: List[str]
+    if isinstance(arg, (int, str)):
+        t = [str(arg)]
+    else:
+        try:
+            t = [str(i) for i in arg]
+        except Exception as e:
+            raise TypeError(
+                "Argument should be of type int, "
+                f"str or Iterable[int/str], not {type(arg)}."
+            )
+    return t
+
+
+def _api_call(url: str) -> ET.Element:
+    """Use requests to call the vedur.is weather API. Return XML tree."""
     result = requests.get(url)
     if result.status_code != 200:
-        raise RequestException("API status code not 200 OK for URL: {0}".format(url))
+        raise RequestException(f"API status code not 200 OK for URL: {url}")
 
-    return json.loads(result.text)
+    # Remove HTML line breaks (which cause confusion in the XML parsing)
+    t: str = re.sub(r"\s*(<br/>)+\s*", r" ", result.text)
 
+    x_tree = ET.fromstring(t)
+    return x_tree
 
-_OBSERVATIONS_URL = "https://apis.is/weather/observations/{0}?stations={1}"
 
+_OBSERVATIONS_URL: str = (
+    "https://xmlweather.vedur.is/?op_w=xml&type=obs&lang={0}&view=xml"
+    "&ids={1}&params=F;FX;FG;D;T;W;V;N;P;RH;SNC;SND;SED;RTE;TD;R"
+)
 
-def observation_for_station(station_id: int, lang: str = _DEFAULT_LANG) -> Dict:
+
+def observation_for_station(station_ids: _ArgType, lang: str = _DEFAULT_LANG) -> dict:
```

Comment:
Shouldn't this be renamed to observation for stations (i.e. + "s") since it accepts multiple stations? And maybe keep the original observation_for_station as a shim which just calls observation_for_stationS w. a single-item list as arg?

Reply:
Or, alternatively, accept either a list of ints or a single int as arg, with an assert to make sure the fn is being invoked in a sane way?

Reply:
Good idea! Added two wrapper functions with a single ID as an argument.